prow/scripts/create-dns-entry.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+
+#In order to run this script you need to use a service account with DNS Administrator role
+
+set -o errexit
+
+if [ -z "$IP_ADDRESS" ]; then
+    echo "\$IP_ADDRESS is empty"
+    exit 1
+fi
+
+if [ -z "$PROJECT" ]; then
+    echo "\$PROJECT is empty"
+    exit 1
+fi
+
+if [ -z "$DNS_ZONE" ]; then
+    echo "\$DNS_ZONE is empty"
+    exit 1
+fi
+
+if [ -z "$DOMAIN" ]; then
+    echo "\$DOMAIN is empty"
+    exit 1
+fi
+
+RANDOM_STRING=$(cat /dev/urandom | env LC_CTYPE=C tr -dc a-z0-9 | head -c 16; echo)
+DNS_NAME="pull-${RANDOM_STRING}.${DOMAIN}"
+
+gcloud dns --project=${PROJECT} record-sets transaction start --zone=${DNS_ZONE}
+
+gcloud dns --project=${PROJECT} record-sets transaction add ${IP_ADDRESS} --name=${DNS_NAME} --ttl=300 --type=A --zone=${DNS_ZONE}
+
+gcloud dns --project=${PROJECT} record-sets transaction execute --zone=${DNS_ZONE}
+
+SECONDS=0
+END_TIME=$((SECONDS+600)) #600 seconds == 10 minutes
+
+while [ ${SECONDS} -lt ${END_TIME} ];do
```

Comment:
on one of our WG meetings, there was a decision that Go should be our main tools. This script becomes not trivial IMO and we should discuss if it should not be migrated to Go.

Reply:
Agree on complexity. This script should just add DNS entry, it's up to the caller to decide if wait for propagation or not. "Waiting" section can be extracted to another script, or, as Adam suggested we can write a tool for that in go (perhaps such a tool already exists?)